content/en/blog/2024/trustee-deployment.md

```diff
@@ -0,0 +1,351 @@
+commonName_default          = kbs-service
+commonName_max              = 64
+
+[req_ext]
+subjectAltName = @alt_names
+
+[v3_ca]
+subjectAltName = @alt_names
+
+[alt_names]
+DNS.1   = kbs-service
+EOF
+```
+
+Create secret for self-signed certificate:
+
+```bash
+openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout https.key -out https.crt \
+    -config kbs-service-509.conf -passin pass:\
+    -subj "/C=UK/ST=England/L=Bristol/O=Red Hat/OU=Development/CN=kbs-service"
+kubectl create secret generic kbs-https-certificate --from-file=https.crt -n kbs-operator-system
+kubectl create secret generic kbs-https-key --from-file=https.key -n kbs-operator-system
+```
+
+### Trustee ConfigMap object
+
+This command will create the ConfigMap object that provides Trustee all the needed configuration:
+
+```bash
+kubectl apply -f - << EOF
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kbs-config
+  namespace: kbs-operator-system
+data:
+  kbs-config.json: |
+    {
+        "insecure_http" : false,
+        "private_key": "/etc/https-key/https.key",
+        "certificate": "/etc/https-cert/https.crt",
+        "sockets": ["0.0.0.0:8080"],
+        "auth_public_key": "/etc/auth-secret/publicKey",
+        "attestation_token_config": {
+          "attestation_token_type": "CoCo"
+        },
+        "repository_config": {
+          "type": "LocalFs",
+          "dir_path": "/opt/confidential-containers/kbs/repository"
+        },
+        "as_config": {
+          "work_dir": "/opt/confidential-containers/attestation-service",
+          "policy_engine": "opa",
+          "attestation_token_broker": "Simple",
+          "attestation_token_config": {
+            "duration_min": 5
+          },
+          "rvps_config": {
+            "store_type": "LocalJson",
+            "store_config": {
+              "file_path": "/opt/confidential-containers/rvps/reference-values/reference-values.json"
+            }
+          }
+        },
+        "policy_engine_config": {
+          "policy_path": "/opt/confidential-containers/opa/policy.rego"
+        }
+    }
+EOF
+```
+
+### Reference Values
+
+The reference values are an important part of the attestation process. The client collects the measurements from the running software and submits a quote with the claims to the attestation server. These measurements, in order for the attestation protocol to succeed, have to match exactly with the trusted digests that had been registered to Trustee previously. This process guarantees the cVM (confidential VM) is running the expected software stack and that it hasn’t been tampered with.
+
+```bashauthor: "[Suraj Deshmukh](https://linktr.ee/surajssd) & [Pradipta Banerjee](https://www.linkedin.com/in/bpradipt/)"
+kubectl apply -f - << EOF
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: rvps-reference-values
+  namespace: kbs-operator-system
+data:
+  reference-values.json: |
+    [
+    ]
+EOF
+```
```

Comment:
It's a bit unclear why there is an empty list from the statement before, maybe add a sentence to describe what we're seeing here. It might need some

Reply:
added one ref values as an example and provided  a note

Reply:
nit: you might want to pad it so the value is 64 lowercase hex chars

Reply:
> nit: you might want to pad it so the value is 64 lowercase hex chars
@mkulke what do you mean exactly?

Reply:
"1" is not a sha256 hex string, so that would be invalid, I assume.

Reply:
The attestation passes with "1" but it doesn't with "6b86b273ff34fce19d6b804eff5a3f5747ada4eaa22f1d49c01e52ddb7875b4b" that should be the sha256 value. Any clues?

Reply:
if you run kbs w/ RUST_LOG=debug set you should be able to see the output of the claims map and the reference value map, maybe this is just an oddity of the sample provider (and we should fix the attester to produce a sha256 value or pad it int the sample verifier).

Reply:
This is what I got when enabling the debug log:
```
[2024-06-12T07:51:43Z DEBUG attestation_service] flattened_claims: {
        "init_data": String(""),
        "report_data": String("p2xcKTRVU95NAW6vQNGqphcAYcRhv8SsXmjX1BjHfFNp5/O1ESp5v0oxWw1+VUcg"),
        "sample.svn": String("1"),
    }
```

Adding disclaimer note about the sample TEE

Reply:
ah `svn` is "security version number" afaik, so it's not a measurement and that might be confusing, since the sample attester doesn't provide any measurements. Talking about measurements and reference values doesn't really make sense here, maybe? Since you already have to push a new policy, why not write a policy that checks the value instead? like: "tee has to be sample, and sample.svn must be >= 1". what do you think?

Reply:
AFAIK the claims come as a list of key/value items and  every HW TEE platform has a proprietary list of keys (see [parsed_claims)](https://github.com/confidential-containers/trustee/blob/main/attestation-service/docs/parsed_claims.md).
Isn't _sample.svn_  following the same path/logic of real claims? (even though just for testing purposes).
What I'm saying is that for real attesters you would follow the same approach/instructions as the method provided for the sample attester.
I thought this could be a value added for the blog, but if in your opinion is not accurate I'll remove it

Reply:
the RVPS is really only meant for measurement hashes as the schema indicates, anything else would have to be verified by the policy. that you can verify sample.svn (a number, not a sha256) via RVPS isn't probably intended by design. I wouldn't advertise a usage like this, since the RVPS is in the process of [being redesigned](https://github.com/confidential-containers/trustee/issues/350) to cover more complex provenance data, so that blog post would probably not age well.

Instead of using the RVPS in the post, I would suggest to use a sample rego policy, replacing the sed expression that we have at the end with a full policy body that checks for the svn value. That might be more useful for readers.

Reply:
Thanks for the suggestion. Resource policy updated accordingly